### ISSUE TITLE
Architecture of the game data json

### ISSUE THREAD

**Issue by TheCheif** [CodingTrain/Oregon-Trail#21]
Hi,
i would like to propose a structure for the game data json file, which is called "slide_data.json" in the repository. 

At the JSON root you have a start and end name as Entry- and Exitpoint for the game. Then you have a object called Pages, which holds all the Data for a given Page. Each Page has a Name to reference it in other Pages (think of it as an ID) and a type. The type determines, which Class should Render the Page. e.g. a Page with the Type "shop" should be rendered by the shop class. Everything the class would need to do it's job, would then be in the data object for the page. So when instantiating the Class, you would give it the data Object and it would know how to deal with this data. 

This approach would be totally data driven. The type Classes would be instantiated with the data, and some other Class, propably a game Manager of some sort, would get the Instance and would go from there. You propably would need more than one class per Type, for the different game Mechanics, e.g. one for Rendering, one for taking over the input and so on. 

The JSON could look something like this

```json
{
    "start": "startPage",
    "end": "ending",
    "pages": [{
        "name": "startPage",
        "type": "start",
        "data": {
            "title": "The Oregon Trail",
            "choices": [{
                "text": "Travel the trail",
                "action": "changePage",
                "newPageName": "travel",
            }, {
                "text": "Learn about the trail",
                "action": "changePage",
                "newPageName": "learn",
            }, {
                "text": "See the Oregon Top Ten",
                "action": "changePage",
                "newPageName": "topTen",
            }, {
                "text": "Turn sound off",
                "action": "toggleSound",
                "atlernativeText": "Turn sound on"
            }],
            "prompt": "What is your choice?",
            "img": "img/title.png",
        }
    }, {
        "type": "choice",
        "name": "travel",
        "data": {
            "text": "First text",
            "choices": [...],
            "img": "",
            "nextPage": "story1"
        }
    }, {
        "type": "story",
        "name": "story1",
        "data": {
            "texts": [
                "This is the first text",
                "This is the second text, which will appear after pressing space"
            ]
        }
    }, {
        "type": "shop",
        "name": "firstShop",
        "data": {
            "items": [{
                "id": 1,
                "name": "Some thing to buy",
                "description": "This is a buyable thing",
                "price": 1600
            }],
            "topText": "Here could be your shop text",
            "nextPage": "pageAfterShop"
        }
    }]
}
```

**Comment by Rodak123** [comment]
Maybe each page would extend the core `Page` class with added functionality.

**Comment by shiffman** [comment]
I love this idea! Do you want to take a stab at implementing a "hello, world" version of this with just 1-3 pages (like you proposed) or shall I take the next steps and adapt your sample from here? Either way works for me! At the moment, I'm working on the user keyboard input.

**Comment by shiffman** [comment]
I think this can be closed now that #31 has been merged, we can open new issues as things come up!
